docs/source/fault_tolerance/usage_guide.rst

```diff
@@ -55,15 +55,38 @@ using ``--ft-<parameter-name>``.
 If ``--max-restarts`` is specified, the launcher restarts failed workers.
 The restart behavior depends on the ``--ft-restart-policy`` parameter, which supports two modes:
 
-* ``any-failed`` (default)  
-  All workers are restarted if any worker fails.  
+* ``any-failed`` (default)
+  All workers are restarted if any worker fails.
 
 * ``min-healthy``
-  Workers are restarted when the number of healthy nodes (nodes where all worker processes are running) 
-  falls below the minimum specified in ``--nnodes``. This allows for some worker failures to be handled 
+  Workers are restarted when the number of healthy nodes (nodes where all worker processes are running)
+  falls below the minimum specified in ``--nnodes``. This allows for some worker failures to be handled
   without restarting remaining workers, e.g., with the :doc:`../inprocess/index`.
   For details on how ``min-healthy`` policy interacts with :doc:`../inprocess/index` see :doc:`integration/inprocess`.
 
+GPU Memory Reclaim
+^^^^^^^^^^^^^^^^^^
+
+When ``--max-restarts`` is specified, ``ft_launcher`` can optionally wait for GPU memory to be
+released before starting new workers after a restart. This helps ensure that GPU memory from
+terminated workers has been fully reclaimed before starting new processes.
+
+This feature is controlled by three parameters:
+
+* ``--ft-gpu-memory-reclaim-timeout`` (default: 50.0 seconds)
+  Timeout for waiting for GPU memory to drop below the tolerance threshold. Set to 0 to disable the feature.
+
+* ``--ft-gpu-memory-tolerance-mb`` (default: 512.0 MB)
+  Maximum allowed GPU memory usage. The launcher waits until GPU memory drops below this threshold.
```

Comment:
Can the launcher also try to allocate and free memory, say 40-80 GB, on the GPUs to kickstart reclaim and ensure allocations succeed?